File: StorageRW/app/src/main/java/com/example/storagerw/MainActivity.java
package com.example.storagerw;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import java.io.BufferedReader;
import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.InputStreamReader;

public class MainActivity extends AppCompatActivity {
    EditText inputField;
    Button writeButton;
    TextView displayText;
    Button readbutton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        inputField = findViewById(R.id.editText);
        writeButton = findViewById(R.id.saveButton);
        displayText = findViewById(R.id.diplayText);
        readbutton = findViewById(R.id.readButton);

        writeButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                writeFile();
            }
        });

        readbutton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                readFile();
            }
        });

    }
    public void writeFile(){
        String texttosave = inputField.getText().toString();

        try{
            FileOutputStream fileOutputStream = openFileOutput("SDL File.txt", MODE_PRIVATE);
            fileOutputStream.write(texttosave.getBytes());
            fileOutputStream.close();

            Toast.makeText(getApplicationContext(), "Text saved successfully", Toast.LENGTH_SHORT).show();
            inputField.setText("");
        }catch (FileNotFoundException e){
            e.printStackTrace();
        }catch (IOException e){
            e.printStackTrace();
        }
    }
    public void readFile(){
        try {
            FileInputStream fileInputStream = openFileInput("SDL File.txt");
            InputStreamReader inputStreamReader = new InputStreamReader(fileInputStream);

            BufferedReader bufferedReader = new BufferedReader(inputStreamReader);
            StringBuffer stringBuffer = new StringBuffer();

            String Lines;
            while((Lines = bufferedReader.readLine()) !=null){
                stringBuffer.append(Lines+"\n");
            }
            displayText.setText(stringBuffer.toString());

        }catch (FileNotFoundException e){
            e.printStackTrace();
        }catch (IOException e){
            e.printStackTrace();
        }
    }
}

File: AboutMe/app/src/main/java/com/example/aboutme/MainActivity.kt
package com.example.aboutme

import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.view.inputmethod.InputMethodManager
import android.widget.Button
import android.widget.EditText
import android.widget.TextView

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        findViewById<Button>(R.id.done_button).setOnClickListener {
            addNickname(it)
        }
        findViewById<TextView>(R.id.nickname_text_view).setOnClickListener {
            updateNickname(it)
        }

    }



    private fun addNickname(view : View){
        val editText = findViewById<EditText>(R.id.nickname_edit_text)
        val nickNameTextView = findViewById<TextView>(R.id.nickname_text_view)
        nickNameTextView.text = editText.text
        editText.visibility = View.GONE
        view.visibility = View.GONE
        nickNameTextView.visibility = View.VISIBLE

        val inputMethodManager = getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
        inputMethodManager.hideSoftInputFromWindow(view.windowToken, 0)
    }

    private fun updateNickname(view: View){
        val editText = findViewById<EditText>(R.id.nickname_edit_text)
        val button_nick_name = findViewById<Button>(R.id.done_button)
        editText.visibility = View.VISIBLE
        button_nick_name.visibility = View.VISIBLE
        view.visibility = View.GONE

        editText.requestFocus()

        val imm = getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
        imm.showSoftInput(editText, 0)
    }
}